components/tools/OmeroPy/src/omero/plugins/metadata.py

```diff
@@ -622,7 +621,9 @@ def pixelsize(self, args):
             if args.z:
                 pixel.setPhysicalSizeZ(omero.model.LengthI(args.z, unit))
 
-        client.getSession().getUpdateService(ctx).saveCollection(pixels)
+        groupId = pixels[0].getDetails().getGroup().getId().getValue()
+        ctx = {'omero.group': str(groupId)}
```

Comment:
conn.SERVICE_OPTS.setOmeroGroup(str(groupId)) is another option

Reply:
For some reason that doesn't work. If I set `conn.SERVICE_OPTS.setOmeroGroup(str(groupId))` instead of using the `ctx`, I'll get a `ome.conditions.SecurityViolation` when running the command as non-owner of the image.
Same further up: `pixels = conn.getQueryService().findAllByQuery(q, params, ctx)` works, but when setting `conn.SERVICE_OPTS` instead of using the `ctx`, it doesn't.